### DISCUSSION TITLE
Faststream CLI workers and logging

### DISCUSSION THREAD

**Discussion by Jonas1312** [ag2ai/faststream#1593] [Q&A]
Hi,

I have some questions about the faststream CLI.

It seems like when setting a number of `workers `greater than 1, the CLI will call the class `Multiprocess(BaseReload)`, which inherits from `BaseReload`. This enables the workers to be automatically reloaded if they crash. In my case, I have some issues with my nats server sometimes, and the natsbroker sometimes crashes because of an `nats.errors.TimeoutError`. I noticed that the faststream app would auto-restart, which is great.

Is this feature (worker auto-restart) also available when using only 1 worker (that is one process)? If not, does this mean that the main/only process will crash and not be restarted?

Another question about the logging through the CLI. When using` faststream run app --workers 5` , if my natsbroker fails to connect, I have those logs in stderr:

![image](https://github.com/user-attachments/assets/2a4e21d4-9d31-454a-ae8b-59f24daedf6e)

The problem is that they are not json formatted, and datadog cannot parse them properly.

I already modify  the formatters of those logger:
- from nats.aio.client import _logger as nats_logger
- from faststream.log import logger as faststream_logger
- logging.getLogger("faststream.access.nats")

But seems like it's not enough. Is there a way to format the log properly when the worker crashes?

Thanks!

**Comment by Lancetnik** [comment]
FastStream supports connection recovery without multiprocessing. It is the Broker feature.
To use JSON formatted logs u should replace FastStream and Broker logger object as u can see in the doc example: https://faststream.airt.ai/latest/getting-started/logging/

**Reply by Jonas1312** [reply]
I have already passed my own loggers to the faststream app and natsbroker, as written in my post.

It works in datadog, you can see the "processed" and "received" logs of faststream (https://github.com/airtai/faststream/blob/main/faststream/broker/middlewares/logging.py):

![image](https://github.com/user-attachments/assets/5c511a52-d6f3-460e-894f-974a4ae8681d)

However, whenever the nats library throws a nats.errors.TimeoutError, the child process crashes and is restarted by faststream.
